pkg/host/common.go

```diff
@@ -0,0 +1,267 @@
+package host
+
+import (
+	"context"
+	"crypto/ed25519"
+	"errors"
+	crypto "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	"github.com/libp2p/go-libp2p-core/routing"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	ps "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-msgio"
+	"github.com/sonr-io/sonr/pkg/config"
+	t "go.buf.build/grpc/go/sonr-io/motor/core/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+// Config returns the configuration of the node
+func (n *hostImpl) Config() *config.Config {
+	return n.config
+}
+
+// Host returns the host of the node
+func (hn *hostImpl) Host() host.Host {
+	return hn.host
+}
+
+// HostID returns the ID of the Host
+func (n *hostImpl) HostID() peer.ID {
+	return n.host.ID()
+}
+
+// Ping sends a ping to the peer
+func (n *hostImpl) Ping(pid string) error {
+	return nil
+}
+
+// PrivateKey returns the private key of the node
+func (n *hostImpl) PrivateKey() (ed25519.PrivateKey, error) {
+	// Get Raw Private Key
+	buf, err := n.privKey.Raw()
+	if err != nil {
+		return nil, err
+	}
+	return ed25519.PrivateKey(buf), nil
```

Comment:
does this mean that the host can only support ed25519 keys?

Reply:
Yes for now - out of scope